rocky/rocky/views/mixins.py

```diff
@@ -99,11 +99,11 @@ def get_origins(
         self,
         reference: Reference,
         organization: Organization,
-    ) -> tuple[list[OriginData], list[OriginData], list[OriginData]]:
+    ) -> dict[str, list[OriginData]]:
```

Comment:
Although this is already a nice improvement, I think we can go one step further by introducing a typed dict. This will improve readability and allows code to be type checked more effectively. e.g.:
```python3
from typing import TypedDict

class OriginsResult(TypedDict):
    declarations: list[OriginData]
    observations: list[OriginData]
    inferences: list[OriginData]

def get_origins(...) -> OriginsResult: ...
```

This only alters the return type, the `return` statement won't change. Not sure if `OriginsResult` is the best name, but it's a good suggestion to begin with.

Reply:
Agreed will implement, naming is always the hard part.